go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2286,15 +2297,16 @@ func newGateway(
 			}
 		}
 
-		gw.openflowManager, err = newGatewayOpenFlowManager(gwBridge, exGwBridge, subnets, nodeIPs)
+		gw.openflowManager, err = newGatewayOpenFlowManager(gwBridge, exGwBridge, subnets, nodeIPs, watchFactory)
 		if err != nil {
 			return err
 		}
 
 		// resync flows on IP change
 		gw.nodeIPManager.OnChanged = func() {
 			klog.V(5).Info("Node addresses changed, re-syncing bridge flows")
-			if err := gw.openflowManager.updateBridgeFlowCache(subnets, gw.nodeIPManager.ListAddresses(), gw.isPodNetworkAdvertised); err != nil {
+			if err := gw.openflowManager.updateBridgeFlowCache(subnets, gw.nodeIPManager.ListAddresses(),
+				gw.isPodNetworkAdvertised, false); err != nil {
```

Comment:
So it make sense to remove `gw.isPodNetworkAdvertised` from here and pass it instead in `gatewayInitInternal.bridgeForInterface` instead to initialize the proper value for the `bridgeUDNConfiguration` there